### ISSUE TITLE
Improve docs to reduce confusion about static frameworks and static libraries

### ISSUE THREAD

**Issue by paulb777** [CocoaPods/CocoaPods#7579]
There is confusion in the community about the differences between static frameworks and static libraries and what's available with 1.4.0 and 1.5.0.  An example is https://github.com/apple/swift-protobuf/issues/723 where the OP mixes up the two concepts. The podspec documentation for `static_framework` is very light.

The docs should be improved to reduce that confusion. 

Part of the answer is:

he best explanation of static frameworks is probably in the [1.4.0 CocoaPods blog announcement](http://blog.cocoapods.org/CocoaPods-1.4.0/).  Swift static libraries just released in the first 1.5.0 beta and the docs still need to be written.

In CocoaPods, a big difference between static frameworks and static libraries is that static frameworks are controlled by the podspec and static libraries by the Podfile.   Until CocoaPods 1.5.0, static libraries did not support modules and were therefore limited to Objective C apps.  With 1.5.0, `use_frameworks!` is no longer required for Swift apps and the pods will be built as static libraries.

For most pods, there is likely no reason to add static_framework's, since it is better to give the static versus dynamic control to the developer in their Podfile. However, if a pod has dependencies on static frameworks (most binary `vendored_framework` CocoaPods are static frameworks), the static_framework attribute is needed.

**Comment by dnkoutso** [comment]
Agreed.

**Comment by amorde** [comment]
@paulb777 are you planning on tackling this? :)

Another thing I think is worth mentioning in the docs somewhere is the recommended way to "wrap" vendored binaries. Many podspecs are a simple wrapper for a vendored library/framework, and providing steps on how to do that correctly would go a long way

**Comment by paulb777** [comment]
@amorde Agreed that the docs should provide a recommended way to "wrap" vendored binaries. There are a few tutorials, but it would be good to consolidate or recommend the right way from here.  That's probably worth its own issue.

I'm not likely to tackle this anytime soon, but I would have the bandwidth to help and review if anyone else wants to. :P

**Comment by paulb777** [comment]
@maksymmalyhin is working on documentation in https://github.com/firebase/firebase-ios-sdk/pull/5181
